cdap-docs/_common/common-build.sh

```diff
@@ -218,15 +218,22 @@ function build_extras() {
 }
 
 function set_mvn_environment() {
+  check_build_rst
   cd ${PROJECT_PATH}
   if [[ "${OSTYPE}" == "darwin"* ]]; then
     # TODO: hard-coded Java version 1.7
     export JAVA_HOME=$(/usr/libexec/java_home -v 1.7)
   fi
+}
+
+function check_build_rst() {
+  local current_directory=$(pwd)
+  cd ${PROJECT_PATH}
   # check BUILD.rst for changes
   BUILD_RST_PATH="${PROJECT_PATH}${BUILD_RST}"
   test_an_include "${BUILD_RST_HASH}" "${BUILD_RST_PATH}"
   echo
+  cd ${current_directory}
```

Comment:
Except that I find that I don't. When I do a ``cd`` in a function, that **always** changes the current working directory. And when I leave that function and return to the calling function, my current working directory **has changed**. In the function ``set_mvn_environment`` I use that to move to the correct location to call for a build; in ``check_build_rst`` I find I need to reset the location otherwise functions that call this function fail as they are no longer in the correct working directory.

I add some echo statements, commented out that line #236, and found:
```
No includes to be downloaded.
start test_includes pwd: /Users/john/Source/cdap_3.2/cdap-docs/faq
MD5 Hash for BUILD.rst matches
check_build_rst end
end test_includes pwd: /Users/john/Source/cdap_3.2
Error: Cannot find source directory `/Users/john/Source/cdap_3.2/source'.